server/netty-server/loom/src/main/scala/sttp/tapir/server/netty/loom/internal/ws/OxSourceWebSocketProcessor.scala

```diff
@@ -0,0 +1,59 @@
+package sttp.tapir.server.netty.loom.internal.ws
+
+import io.netty.channel.ChannelHandlerContext
+import io.netty.handler.codec.http.websocketx.CloseWebSocketFrame
+import io.netty.handler.codec.http.websocketx.WebSocketCloseStatus
+import io.netty.handler.codec.http.websocketx.WebSocketFrame
+import org.reactivestreams.Processor
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+import ox.*
+import ox.channels.Source
+import sttp.tapir.model.WebSocketFrameDecodeFailure
+import sttp.tapir.server.netty.internal.ws.WebSocketFrameConverters._
+import sttp.tapir.server.netty.loom.OxStreams
+import sttp.tapir.server.netty.loom.internal.ox.OxDispatcher
+import sttp.tapir.server.netty.loom.internal.reactivestreams.OxProcessor
+import sttp.tapir.{DecodeResult, WebSocketBodyOutput}
+
+private[loom] object OxSourceWebSocketProcessor {
+  def apply[REQ, RESP](
+      oxDispatcher: OxDispatcher,
+      pipe: OxStreams.Pipe[REQ, RESP],
+      o: WebSocketBodyOutput[OxStreams.Pipe[REQ, RESP], REQ, RESP, ?, OxStreams],
+      ctx: ChannelHandlerContext 
+  ): Processor[WebSocketFrame, WebSocketFrame] = {
+    val frame2FramePipe: OxStreams.Pipe[WebSocketFrame, WebSocketFrame] = Ox ?=>
+      (source: Source[WebSocketFrame]) => {
+        pipe(
+          source
+            .map { f =>
```

Comment:
won't `.mapAsView` be enough here? We aren't doing any async/blocking operations, so running this on the consumer's thread should be fine?

Reply:
Could you explain which is the consumer thread exactly?

Reply:
whoever runs `.receive`. So yes, it will (transitively) be the thread calling `.foreach`.